terraform/modules/emblem-app/main.tf

```diff
@@ -78,3 +78,102 @@ resource "google_storage_bucket_iam_member" "sessions-iam" {
   role   = "roles/storage.objectAdmin"
   member = "serviceAccount:${google_service_account.website_manager.email}"
 }
+
+############################
+# E2E Delivery testing IAM #
+############################
+
+resource "google_project_iam_member" "delivery_gae_viewer_iam_member" {
+  project  = var.project_id
+  provider = google
+  role     = "roles/appengine.appViewer"
+  member   = "serviceAccount:${data.google_project.ops.number}@cloudbuild.gserviceaccount.com"
+
+  depends_on = [
+    google_project_service.emblem_app_services
+  ]
+}
+
+resource "google_project_iam_member" "delivery_sa_admin_iam_member" {
+  project  = var.project_id
+  provider = google
+  role     = "roles/iam.serviceAccountAdmin"
+  member   = "serviceAccount:${data.google_project.ops.number}@cloudbuild.gserviceaccount.com"
+
+  depends_on = [
+    google_project_service.emblem_app_services
+  ]
+}
+
+resource "google_project_iam_member" "delivery_compute_viewer_iam_member" {
+  project  = var.project_id
+  provider = google
+  role     = "roles/compute.viewer"
+  member   = "serviceAccount:${data.google_project.ops.number}@cloudbuild.gserviceaccount.com"
+
+  depends_on = [
+    google_project_service.emblem_app_services
+  ]
+}
+
+resource "google_project_iam_member" "delivery_iam_reviewer_iam_member" {
+  project  = var.project_id
+  provider = google
+  role     = "roles/iam.securityReviewer"
+  member   = "serviceAccount:${data.google_project.ops.number}@cloudbuild.gserviceaccount.com"
+
+  depends_on = [
+    google_project_service.emblem_app_services
+  ]
+}
+
+resource "google_project_iam_member" "delivery_storage_object_admin_granting_iam_member" {
+  project  = var.project_id
+  provider = google
+  role     = "roles/iam.securityAdmin"
+  member   = "serviceAccount:${data.google_project.target_project.number}@cloudbuild.gserviceaccount.com"
+
+  depends_on = [
+    google_project_service.emblem_ops_services
+  ]
+
+  condition {
```

Comment:
I've only read about, not used this. Really cool! I think a docs link would be valuable here given this is a more advanced technique.

Reply:
Done; I linked to [this page](https://cloud.google.com/iam/docs/conditions-overview).